[flutter_local_notifications] Add CarPlay notification support with IOSInitializationSettings (#2716)
MaikuB/flutter_local_notifications: flutter_local_notifications/README.md

```dart
const AndroidInitializationSettings initializationSettingsAndroid =
    AndroidInitializationSettings('app_icon');
final IOSInitializationSettings initializationSettingsIOS =
    IOSInitializationSettings(
        // Darwin settings
        requestAlertPermission: true,
        requestBadgePermission: true,
        requestSoundPermission: true,
        // iOS-specific settings
        requestCarPlayPermission: true,
    );
final DarwinInitializationSettings initializationSettingsDarwin =
    DarwinInitializationSettings();
final InitializationSettings initializationSettings = InitializationSettings(
    android: initializationSettingsAndroid,
    iOS: initializationSettingsIOS,  // Use iOS-specific settings
    macOS: initializationSettingsDarwin,  // Keep Darwin for macOS
);
```

#### CarPlay Integration

When `requestCarPlayPermission` is set to `true`, the plugin will request CarPlay notification permissions if the device supports it (iOS 10.0+). This allows your app's notifications to appear on the CarPlay interface when the device is connected to a compatible vehicle.

**Important considerations:**

> - CarPlay permissions are only available on iOS 10.0 and later

Reviewer: ```suggestion
- CarPlay permissions are only available on iOS 10.0 or newer
```

For consistency with rest of plugin docs that have been using "or newer"
